packages/mobile/src/store/offline-downloads/sagas/downloadQueueSagas/workers/downloadTrackWorker.ts

```diff
@@ -81,22 +89,78 @@ function* downloadTrackAsync(
     return OfflineDownloadStatus.ERROR
   }
 
-  const trackMetadata: Track & UserTrackMetadata = {
-    ...track,
-    // Empty cover art sizes because the images are stored locally
-    _cover_art_sizes: {}
-  }
-
   try {
     yield* all([
       call(downloadTrackCoverArt, track),
-      call(tryDownloadTrackFromEachCreatorNode, track)
+      call(downloadTrackAudio, track),
+      call(writeTrackMetadata, track)
     ])
-
-    yield* call(writeTrackJson, trackId.toString(), trackMetadata)
   } catch (e) {
     return OfflineDownloadStatus.ERROR
   }
 
   return OfflineDownloadStatus.SUCCESS
 }
+
+function* downloadTrackAudio(track: UserTrackMetadata) {
+  const { track_id, user } = track
+
+  const { creator_node_endpoint } = user
+  const creatorNodeEndpoints = creator_node_endpoint?.split(',')
+  if (!creatorNodeEndpoints) throw new Error('No creator node endpoints')
+
+  const trackFilePath = getLocalAudioPath(track_id)
+  const encodedTrackId = encodeHashId(track_id)
+
+  for (const creatorNodeEndpoint of creatorNodeEndpoints) {
+    const trackAudioUri = `${creatorNodeEndpoint}/tracks/stream/${encodedTrackId}`
+    const response = yield* call(downloadFile, trackAudioUri, trackFilePath)
+    const { status } = response.info()
+    if (status === 200) return
+  }
+
+  throw new Error('Unable to download track audio')
+}
+
+function* downloadTrackCoverArt(track: UserTrackMetadata) {
+  const { cover_art_sizes, cover_art, user, track_id } = track
+  const cid = cover_art_sizes ?? cover_art
+  const imageSources = createAllImageSources({
+    cid,
+    user,
+    size: SquareSizes.SIZE_1000_BY_1000
+  })
+
+  const coverArtUris = imageSources.map(({ uri }) => uri).filter(removeNullable)
+  const covertArtFilePath = getLocalTrackCoverArtDestination(track_id)
+
+  for (const coverArtUri of coverArtUris) {
+    const response = yield* call(downloadFile, coverArtUri, covertArtFilePath)
+    const { status } = response.info()
+    if (status === 200) return
+  }
+
+  throw new Error('Unable to download track cover art')
+}
+
+async function writeTrackMetadata(track: UserTrackMetadata) {
+  const { track_id } = track
+
+  const trackMetadata: Track & UserTrackMetadata = {
+    ...track,
+    // Empty cover art sizes because the images are stored locally
+    _cover_art_sizes: {}
+  }
+
+  const trackMetadataPath = getLocalTrackJsonPath(track_id.toString())
+
+  return await RNFetchBlob.fs.writeFile(
+    trackMetadataPath,
+    JSON.stringify(trackMetadata)
+  )
+}
+
+async function removeDownloadedTrack(trackId: ID) {
+  const trackDirectory = getLocalTrackDir(trackId.toString())
+  return await RNFetchBlob.fs.unlink(trackDirectory)
+}
```

Comment:
Is this all just copied? Any notable changes?

Reply:
its mostly copied with some changes to prevent extra checks on file existence, which should result in speed improvements since half the promises are removed